### PR TITLE
<Dialog/> - added notFullscreenOnMobile prop for mobile

### PR DESCRIPTION
**Pull request by jjaskulis**
# ✨ Pull Request

### 💁 Description

	Added ability to have non-fullscreen modals on mobile
	https://jira.wixpress.com/plugins/servlet/samlsso?redirectTo=%2Fbrowse%2FDSM-1449

...

### 💭 Motivation

Non-fullscreen modal on mobile has vertical constraints -> max height of 440px and minimum top/bottom padding should be 20px. This vertical padding would be needed in edge-cases where screen height would be smaller than possible Dialog component max-height. So to always have vertical padding, added the outer wrapper container for Dialog content.

...

### ☝️ TODOs 

...

### 👀  PR is ready for review 

- [x] Ready for review

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://bo.wix.com/pages/wix-ui-tpa.pr_642/

File: src/components/Dialog/Dialog.visual.tsx [hunk 1]
```diff
@@ -32,15 +33,43 @@ const DialogWithContent = (props) => (
...
+      `Mobile with default props  ${isRtl ? 'rtl' : ''}`,
+      TPAComponentsWrapper({ mobile: true, rtl: isRtl })(<DialogWithContent />),
+    );
+    snap(
+      `Mobile with not fullscreen version  ${isRtl ? 'rtl' : ''}`,
+      TPAComponentsWrapper({ mobile: true, rtl: isRtl })(
+        <DialogWithContent notFullscreenOnMobile />,
+      ),
+    );
   });
```
Comment (sivanharel) [inline src/components/Dialog/Dialog.visual.tsx:63]: Nice!
In order to see how the visual tests story page you could run:
`npm run eyes-storybook-debug`

When you run it you would be able to see that the mobile tests that you added are not really rendered as they would be on mobile.

We could use the `MobileExample.tsx` file in the visual tests in order to use the mobile iframe. (this is the mobile iframe that we use on the Storybook).
https://github.com/wix/wix-ui-tpa/blob/master/stories/utils/MobileExample.tsx

In order to import this file from the `Dialog.visual.tsx` file we should add require in the `.storybook-test/config.js`.
https://github.com/wix/wix-ui-tpa/blob/819117845ec1a142103818720a1dd7f4cc0e473a/.storybook-test/config.js#L7
` require('stories/utils/MobileExample');`

File: src/components/Dialog/Dialog.st.css [hunk 2]
```diff
@@ -57,6 +61,24 @@
...
+    height: calc(100% - 2 * value(NotFullscreenMobilePadding));
+}
+
+.root:mobile:notFullscreenMobile .contentWrapper{
+    position: relative;
+    height: auto;
+    max-height: 440px;
+    min-width: auto;
+}
+
```
Comment (sivanharel) [inline src/components/Dialog/Dialog.st.css:81]: We might want to update the inner padding on mobile mode.
Currently this is the padding value:
`.dialogContent {
  padding: 80px 32px;
}`

I will ask Michal (our designer about it).

**Review by sivanharel — COMMENTED** [review]
Very nice!!!
I wrote some small comments

File: src/components/Dialog/Dialog.st.css (hunk 2 above)
Comment (jjaskulis) [reply src/components/Dialog/Dialog.st.css:81]: all right, will update if needed :)

File: src/components/Dialog/Dialog.visual.tsx (hunk 1 above)
Comment (jjaskulis) [reply src/components/Dialog/Dialog.visual.tsx:63]: ah nice, I saw that it's not really rendered as mobile, checked some components and didn't find anything I could use to simulate mobile properly. Glad you showed this, I'll be able to add another test case to simulate edge case with small mobile screen height.

File: src/components/Dialog/docs/examples.ts [hunk 3]
```diff
@@ -71,3 +71,28 @@ export const mobileExample = buildExample(`
       </Dialog>
      </MobileExample>
 `);
+
+export const mobileExampleNotFullscreen = buildExample(`
+    <MobileExample>
+      <Button upgrade onClick={this.onOpenDialogButtonClick}>Open Dialog</Button>
+      <Dialog isOpen={isDialogOpen} onClose={this.onCloseDialog} notFullscreenOnMobile>
+        <div className="content" style={{ textAlign: 'center' }}>
+            <Text typography="MobileSmallTitleFont" tagName="div">Discard draft?</Text>
```
Comment (sivanharel) [inline src/components/Dialog/docs/examples.ts:80]:
```suggestion
            <Text typography="largeTitle" tagName="div">Discard draft?</Text>
```

I know that you just copied it from the other example but I found a bug here. `MobileSmallTitleFont` is not a valid value for `typography` and should be `largeTitle`.

Comment (sivanharel) [reply src/components/Dialog/docs/examples.ts:80]: Would be happy if you could change it in the other example as well.

File: src/components/Dialog/docs/examples.ts [hunk 4]
```diff
@@ -71,3 +71,28 @@ export const mobileExample = buildExample(`
...
+            <div className="text-container" style={{ marginTop: '16px', marginBottom: '32px' }}>
+                <Text typography="MobileRunningTextFont" tagName="div">Are You Sure you want to discard the changes you made?</Text>
```
Comment (sivanharel) [inline src/components/Dialog/docs/examples.ts:82]:
```suggestion
                <Text typography="runningText" tagName="div">Are You Sure you want to discard the changes you made?</Text>
```

File: src/components/Dialog/docs/examples.ts [hunk 5]
```diff
@@ -71,3 +71,28 @@ export const mobileExample = buildExample(`
...
+            </div>
+            <div className="primary-btn-container" style={{marginBottom: '8px'}}>
+                  <Button upgrade>
+                    PRIMARY
+                  </Button>
+            </div>
+            <Button
+              upgrade
+              priority={PRIORITY.basicSecondary}
+            >
```
Comment (sivanharel) [inline src/components/Dialog/docs/examples.ts:92]:
```suggestion
               style={{ width: '100%', boxSizing: 'border-box' }}
            >
```

File: src/components/Dialog/docs/examples.ts [hunk 6]
```diff
@@ -71,3 +71,28 @@ export const mobileExample = buildExample(`
...
+    <MobileExample>
+      <Button upgrade onClick={this.onOpenDialogButtonClick}>Open Dialog</Button>
+      <Dialog isOpen={isDialogOpen} onClose={this.onCloseDialog} notFullscreenOnMobile>
+        <div className="content" style={{ textAlign: 'center' }}>
+            <Text typography="MobileSmallTitleFont" tagName="div">Discard draft?</Text>
+            <div className="text-container" style={{ marginTop: '16px', marginBottom: '32px' }}>
+                <Text typography="MobileRunningTextFont" tagName="div">Are You Sure you want to discard the changes you made?</Text>
+            </div>
+            <div className="primary-btn-container" style={{marginBottom: '8px'}}>
+                  <Button upgrade>
```
Comment (sivanharel) [inline src/components/Dialog/docs/examples.ts:85]:
```suggestion
                  <Button upgrade style={{ width: '100%', boxSizing: 'border-box' }}>
```

File: src/components/Dialog/docs/examples.ts [hunk 7]
```diff
@@ -52,18 +52,44 @@ export const mobileExample = buildExample(`
...
+                <Text typography="runningText" tagName="div">Are You Sure you want to discard the changes you made?</Text>
+            </div>
+            <div className="primary-btn-container" style={{marginBottom: '8px'}}>
+                  <Button upgrade style={{ width: '100%', boxSizing: 'border-box' }}>
+                    PRIMARY
+                  </Button>
+            </div>
+            <Button
+              upgrade
+              priority={PRIORITY.basicSecondary}
```
Comment (sivanharel) [inline src/components/Dialog/docs/examples.ts:66]:
```suggestion
              priority={PRIORITY.basicSecondary}
              style={{ width: '100%', boxSizing: 'border-box' }}
```

**Review by sivanharel — COMMENTED** [review]
Awesome!!
Very nice solution!
I just have one question :)

File: src/test/visualTestWrapper.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
+import * as React from 'react';
+import { TPAComponentsProvider } from '../components/TPAComponentsConfig';
+import { MobileExample } from '../../stories/utils/MobileExample';
+
+export class AsyncVisualTestWrapper extends React.Component<{
+  isMobile: boolean;
+  isRtl: boolean;
+  onDoneCallback(): void;
+  onDoneTimeout?: number;
```
[the remark this answers is not in the document]
Comment (jjaskulis) [reply src/test/visualTestWrapper.tsx:9]: was testing different timeouts, but yeah not using it anywhere. can remove

**Review by sivanharel — APPROVED** [review]
